cni/network/network.go

```diff
@@ -1404,27 +1418,38 @@ func convertNnsToIPConfigs(
 	return ipConfigs
 }
 
-func convertInterfaceInfoToCniResult(info network.InterfaceInfo, ifName string) *cniTypesCurr.Result {
-	result := &cniTypesCurr.Result{
-		Interfaces: []*cniTypesCurr.Interface{
-			{
+func (plugin *NetPlugin) convertInterfaceInfoToCniResult(info IPAMAddResult, ifName string) *cniTypesCurr.Result {
+	result := &cniTypesCurr.Result{}
+
+	for key := range info.interfaceInfo {
+		interfaceInfo := info.interfaceInfo[key]
+
+		if interfaceInfo.NICType == cns.BackendNIC {
+			// get an updated PnP Device ID(PciID)
+			pnpDeviceID, _ := plugin.nm.GetPnPDeviceID(interfaceInfo.PnPID)
+			result.Interfaces = append(result.Interfaces, &cniTypesCurr.Interface{
+				Name:  "ib" + key,
+				Mac:   interfaceInfo.MacAddress.String(),
+				PciID: pnpDeviceID,
+			})
+		} else if interfaceInfo.NICType == cns.InfraNIC {
+			result.Interfaces = append(result.Interfaces, &cniTypesCurr.Interface{
 				Name: ifName,
-				Mac:  info.MacAddress.String(),
-			},
-		},
-		DNS: cniTypes.DNS{
-			Domain:      info.DNS.Suffix,
-			Nameservers: info.DNS.Servers,
-		},
-	}
-
-	if len(info.IPConfigs) > 0 {
-		for _, ipconfig := range info.IPConfigs {
-			result.IPs = append(result.IPs, &cniTypesCurr.IPConfig{Address: ipconfig.Address, Gateway: ipconfig.Gateway})
-		}
+			})
+			result.DNS = cniTypes.DNS{
+				Domain:      interfaceInfo.DNS.Suffix,
+				Nameservers: interfaceInfo.DNS.Servers,
+			}
+
+			if len(interfaceInfo.IPConfigs) > 0 {
+				for _, ipconfig := range interfaceInfo.IPConfigs {
+					result.IPs = append(result.IPs, &cniTypesCurr.IPConfig{Address: ipconfig.Address, Gateway: ipconfig.Gateway})
```

Comment:
change should be only for ib nic for adding pnp id and rest of the code should be same.. why else if change required here

Reply:
because we need to give containerd cniResult to containerd with only infraNIC and IBNIC; if other types of NICs come then we should skip them